### DISCUSSION TITLE
X-lab 参与度与贡献度的游戏化运营提议

### DISCUSSION THREAD

**Discussion by will-ww** [X-lab2017/open-wonderland#49] [2️⃣ 聊项目]
Hello all，

Frank 的创新工作有了些进展。完成了[ sealos 社区](https://github.com/labring/sealos)项目内的价值网络计算，下图是他们项目的上月计算结果，并且已经分配上链。目前每个月分配出去约 5 千，一年内预计可以到每月分配 5W 左右，会使用 OpenRank 作为分配算法，结果也高度符合他的预期。

<img width="432" alt="c8b85aeba6717a2eb7a23fb0ecf5c0d" src="https://user-images.githubusercontent.com/15010826/170393804-e93b8d6d-cb21-4841-8191-7aef257634b4.png">

这项工作已经完全自动化了，每个月初自动结算上月的结果并自动化链上分配，而且 Frank 建议他们社区开始大量使用 GitHub reactions 来做日常化的 peer review 中的态度表达，这些数据后续也会加入到网络中。

Frank 同时也跑了一下 OpenDigger 2022.3 月份的数据：

<img width="471" alt="44776b7313d656dbebc51ece940e3c5" src="https://user-images.githubusercontent.com/15010826/170394844-43d33696-5edc-4d3c-940c-5dd7d13f02a0.png">

初步探讨了下：X-lab 也可以开始尝试，数据本身也都导进去了，不过目前就是每个月仓库独立来做，仓库之间的关联是不考虑的。跨仓的分配虽然比单仓复杂一些，需要考虑仓库之间的相对价值，否则无法确定在哪个仓库上分配多少的问题。

但这个其实就是全域的 OpenRank 要解决的问题。用全域 OpenRank 先解决每个仓库的相对价值比较，然后再用单仓 OpenRank 来解决每个仓库内的分配问题，这个是一套东西。多仓的话其实纯走累加是没问题的。

大家可以一起来设计几个问题：

- 从 6 月份开始，我们先尝试起来，用一个月的实践迭代；7月份正式运行；
- 奖金分配上，学院这边的配置，例如拿出 5000元/月，进行分配；我和 Frank 也可以从其他企业项目中拿出一部分；
- 需要设计参加仓库的范围、人员、统计机制、反馈方式（现金、礼券、奖品等）等，同时给出一些指导性文件；
- 作为一个鼓励性手段，而非强制性，可以选择性参加。

大家看看，有啥好的想法，可以随时提出~ 

@X-lab2017/x-lab

**Comment by xgdyp** [comment]
太牛了！！



---- 回复的原邮件 ----
| 发件人 | Will ***@***.***> |
| 日期 | 2022年05月26日 09:32 |
| 收件人 | ***@***.***> |
| 抄送至 | ***@***.***>、Team ***@***.***> |
| 主题 | [X-lab] X-lab 社区参与度与贡献度的游戏化运营提议 (Discussion #49) |

Hello all，

Frank 的创新工作有了些进展。完成了 sealos 社区项目内的价值网络计算，下图是他们项目的上月计算结果，并且已经分配上链。目前每个月分配出去约 5 千，一年内预计可以到每月分配 5W 左右，会使用 OpenRank 作为分配算法，结果也高度符合他的预期。

这项工作已经完全自动化了，每个月初自动结算上月的结果并自动化链上分配，而且 Frank 建议他们社区开始大量使用 GitHub reactions 来做日常化的 peer review 中的态度表达，这些数据后续也会加入到网络中。

Frank 同时也跑了一下 OpenDigger 2022.3 月份的数据：

初步探讨了下：X-lab 也可以开始尝试，数据本身也都导进去了，不过目前就是每个月仓库独立来做，仓库之间的关联是不考虑的。跨仓的分配虽然比单仓复杂一些，需要考虑仓库之间的相对价值，否则无法确定在哪个仓库上分配多少的问题。

但这个其实就是全域的 OpenRank 要解决的问题。用全域 OpenRank 先解决每个仓库的相对价值比较，然后再用单仓 OpenRank 来解决每个仓库内的分配问题，这个是一套东西。多仓的话其实纯走累加是没问题的。

大家可以一起来设计几个问题：

从 6 月份开始，我们先尝试起来，用一个月的实践迭代；7月份正式运行；
奖金分配上，学院这边的配置，例如拿出 5000元/月，进行分配；我和 Frank 也可以从其他企业项目中拿出一部分；
需要设计参加仓库的范围、人员、统计机制、反馈方式（现金、礼券、奖品等）等，同时给出一些指导性文件；
作为一个鼓励性手段，而非强制性，可以选择性参加。

大家看看，有啥好的想法，可以随时提出~

@X-lab2017/x-lab

**Reply by wangleo61** [reply]
这个太有意思啦

**Comment by bifenglin** [comment]
这个好哇，调动大家积极性，也符合开源理念，也可以验证一下frank的开源项目内的价值网络计算。

**Comment by birdflyi** [comment]
牛！！！贡献值量化了~
ratio的格式似乎还需要调整？都是0.xxx%感觉有点怪，公式应该是`[open_rank/SUM(open_rank)]×100%`。
其他想法：单仓的及时反馈还是有利于激发积极性的，不过考虑跨仓情况下，单个项目的总open_rank值`SUM(open_rank)`是不是还需要引入平滑（如移动平均法Moving average，指数平滑法Exponential Smoothing）来调控突发事件带来的风险呢？

**Comment by AliceCodeZhang** [comment]
赞！ 建议大家都参加，设置一个试运行期限，然后运行结束之后复盘讨论一下

**Reply by will-ww** [reply]
> 赞！ 建议大家都参加，设置一个试运行期限，然后运行结束之后复盘讨论一下

那就成了 **X-lab Summer of Code**（XSoC）了~ :-)

**Reply by Wuhanqin** [reply]
之前帮助过类似的u-ospo做过贡献，好像上游负责的学生没有管我的回复啊，嘤嘤嘤。

**Comment by cleare630** [comment]
很棒的project，对分析洞察开源项目很有帮助

**Comment by Wuhanqin** [comment]
老师，针对apacheanswer这样类似的运营问答的平台，有没有评估这个问答帖子博客这样的贡献度量化算法？

**Comment by Wuhanqin** [comment]
建立这个机制，最简单的方式就是模仿百度飞桨，直接做任务搞积分，然后给礼物，送人才认证和贡献者激励证书这样的，其实很多社区的鼓励方式大同小异。可以模仿。建议做一个先驱。
